controllers/disruption_controller.go

```diff
@@ -272,6 +329,97 @@ func (r *DisruptionReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	return ctrl.Result{}, nil
 }
 
+// createInstanceSelectorCache creates this instance's cache if it doesn't exist and attaches it to the controller
+func (r *DisruptionReconciler) manageInstanceSelectorCache(instance *chaosv1beta1.Disruption) error {
+	if !instance.Spec.DynamicTargeting {
+		return nil
+	}
+
+	disNamespacedName := types.NamespacedName{Name: instance.Name, Namespace: instance.Namespace}
+	disCompleteSelector, err := targetselector.GetLabelSelectorFromInstance(instance)
+
+	if err != nil {
+		return fmt.Errorf("error getting instance selector: %w", err)
+	}
+	// if it doesn't exist, create the cache context to re-trigger the disruption
+	if r.CachesCancel[disNamespacedName] == nil {
```

Comment:
We run the teeny tiny risk here that a Disruption can be deleted without the proper teardown, then re-created with the same name but a different spec, and that could screw up the cache. I dont have a suggested fix, it's extremely unlikely anyway. Maybe just leave a comment that we are accepting that failure case?

Reply:
Map key is now a hash of the disruption's `Spec` (`Count` excluded, like the admission hook) + Namespaced Name. It  excludes the scenario you mentionned.
I imagine it is still possible to get something messy if a single disruption is manually pulsed and a race condition occurs on the cache deletion/creation, and I'll set up a monitor for such a leak.